modin/pandas/test/utils.py

```diff
@@ -1244,23 +1244,35 @@ def teardown_test_files(test_paths: list):
         teardown_test_file(path)
 
 
-def sort_index_for_equal_values(series, ascending=False):
-    if series.index.dtype == np.float64:
+def sort_index_for_equal_values(df, ascending=True):
+    """Sort `df` indices of equal rows."""
+    if df.index.dtype == np.float64:
         # HACK: workaround for pandas bug:
         # https://github.com/pandas-dev/pandas/issues/34455
-        series.index = series.index.astype("str")
-    res = series.groupby(series, sort=False).apply(
+        df.index = df.index.astype("str")
+    res = df.groupby(by=df if df.ndim == 1 else df.columns, sort=False).apply(
         lambda df: df.sort_index(ascending=ascending)
     )
-    if res.index.nlevels > series.index.nlevels:
+    if res.index.nlevels > df.index.nlevels:
         # Sometimes GroupBy adds an extra level with 'by' to the result index.
         # GroupBy is very inconsistent about when it's doing this, so that's
         # why this clumsy if-statement is used.
         res.index = res.index.droplevel(0)
-    res.name = series.name
+    # GroupBy discards original index names, so restoring them here
```

Comment:
Why did we do this in the test? Can we do this in the code to make sure this will always pass?

Reply:
removed code regarding to the <i>Series name</i> restoring and suddenly all tests have passed.

As for the <i>index names</i>, we can't do this names restoring in the GroupBy code because its natural behaviour is to set index names with the 'by' labels. Since the logic in this function is just about preprocessing frame before its comparison we want to index names be untouched, that's why we need this names restoring here.

I've clarified this at the in-code comment